processor/transformer/transformer.go

```diff
@@ -375,16 +354,68 @@ func (trans *HandleT) request(ctx context.Context, url string, data []Transforme
 			trans.logger.Errorf("Transformer returned : %v", string(respData))
 			respData = []byte(fmt.Sprintf("Failed to unmarshal transformer response: %s", string(respData)))
 			transformerResponses = nil
-			resp.StatusCode = 400
+			statusCode = 400
 		}
 	}
 
-	if resp.StatusCode != http.StatusOK {
+	if statusCode != http.StatusOK {
 		for i := range data {
 			transformEvent := &data[i]
-			resp := TransformerResponseT{StatusCode: resp.StatusCode, Error: string(respData), Metadata: transformEvent.Metadata}
+			resp := TransformerResponseT{StatusCode: statusCode, Error: string(respData), Metadata: transformEvent.Metadata}
 			transformerResponses = append(transformerResponses, resp)
 		}
 	}
 	return transformerResponses
 }
+
+func (trans *HandleT) doPost(ctx context.Context, rawJSON []byte, url string, tags stats.Tags) ([]byte, int) {
+	var (
+		retryCount int
+		resp       *http.Response
+		respData   []byte
+	)
+	err := backoff.RetryNotify(
```

Comment:
I find the usage of nested retries a bit confusing, and I think it can be avoided if HTTP client errors are handled as transformer unavailability cases. I would also propose to do JSON parsing only if the status code is the expected one.

Reply:
They are retrying for different purposes: transformer unavailability, client errors, issues reading the body are handled by the finite retry mechanism, whereas special error code indicating that backend config is unavailable (not transformer) are handled by the other, infinite retry mechanism.
What could be changed?